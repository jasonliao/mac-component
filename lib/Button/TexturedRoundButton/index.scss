.textured-round-button {
  box-sizing: border-box;
  color: #575757;
  border-width: 1px;
  border-style: solid;
  border-color: #d7d7d7 #d3d3d3 #bfbfbf;
  background: linear-gradient(to bottom, #fefefe, #f3f3f3);
  border-radius: 4px;
  outline: none;
  padding: 1px 14px;
  box-shadow: 0px 0.2px 0.1px 0px rgba(0, 0, 0, 0.15);
}

.textured-round-button:active {
  background: linear-gradient(to bottom, #e5e5e5, #dcdcdc);
  color: #000;
}

.textured-round-button.textured-round-button__disabled,
.textured-round-button.textured-round-button__disabled:active {
  background: #f6f6f6;
  border-width: 1px;
  border-style: solid;
  border-color: #e0e0e0 #dfdfdf #d5d5d5;
  color: #bebebe;
}

.textured-round-button__regular {
  font-size: 13px;
}

.textured-round-button__small {
  font-size: 11px;
}

.textured-round-button__mini {
  font-size: 9px;
}