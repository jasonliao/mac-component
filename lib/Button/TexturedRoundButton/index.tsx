import * as React from 'react'
import classNames from 'classnames'

import Button, { ButtonProps } from '../BaseComponents/Button'

import './index.scss'

export interface TexturedRoundButtonProps extends ButtonProps {}

export default class TexturedRoundButton extends Button<TexturedRoundButtonProps> {
  render() {
    const { size, disabled, style, className, children } = this.props

    const cls = classNames(
      className,
      'textured-round-button',
      `textured-round-button__${size}`,
      {
        'textured-round-button__disabled': disabled,
      }
    )

    return (
      <button
        className={cls}
        style={style}
        onClick={this.handleClick}
      >
        { children }
      </button>
    )
  }
}