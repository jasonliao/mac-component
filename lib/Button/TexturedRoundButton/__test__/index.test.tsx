import * as React from 'react'
import { shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'

import TexturedRoundButton, { TexturedRoundButtonProps } from '../index'

describe('TexturedRoundButton Component', () => {
  const content = 'Textured Round Button'
  it('[UI] render text', () => {
    const wrapper = shallow(<TexturedRoundButton>{ content }</TexturedRoundButton> )

    const text = wrapper.find('button').text()
    expect(text).toBe(content)
  })

  it('[UI] render with mini size', () => {
    const props: TexturedRoundButtonProps = { size: 'mini', }
    const wrapper = shallow(<TexturedRoundButton { ...props } />)

    const hasClass = wrapper.find('.textured-round-button').hasClass(`textured-round-button__${props.size}`)
    expect(hasClass).toBe(true)
  })

  it('[UI] render disabled button', () => {
    const props: TexturedRoundButtonProps = { disabled: true }
    const wrapper = shallow(<TexturedRoundButton { ...props } />)

    const hasClass = wrapper.find('.textured-round-button').hasClass(`textured-round-button__disabled`)
    expect(hasClass).toBe(true)
  })

  it('[BEHAVIOR] button clicked', () => {
    const props: TexturedRoundButtonProps = { onClick: jest.fn() }
    const wrapper = shallow(<TexturedRoundButton { ...props } />)

    wrapper.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })

  it('[BEHAVIOR] disabled button clicked', () => {
    const props: TexturedRoundButtonProps = { disabled: true, onClick: jest.fn() }
    const wrapper = shallow(<TexturedRoundButton { ...props } />)

    wrapper.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(0)
  })

  it('[SNAPSHOT] match the snapshot', () => {
    const props: TexturedRoundButtonProps = {
      size: 'small',
      style: { width: '100px' },
      className: 'custom_classname'
    }

    const tree = renderer.create(<TexturedRoundButton { ...props } >{ content }</TexturedRoundButton>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})