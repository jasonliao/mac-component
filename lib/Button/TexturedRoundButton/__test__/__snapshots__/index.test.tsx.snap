// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TexturedRoundButton Component [SNAPSHOT] match the snapshot 1`] = `
<button
  className="custom_classname textured-round-button textured-round-button__small"
  onClick={[Function]}
  style={
    Object {
      "width": "100px",
    }
  }
>
  Textured Round Button
</button>
`;
